packages/crypto/src/crypto.ts

```diff
@@ -0,0 +1,498 @@
+    const squareRoot = modPow(base, q, p);
+    if ((squareRoot * squareRoot) % p !== base) {
+      throw new Error("could not find a modular square root");
+    }
+    return squareRoot;
+  }
+
+  // Other elliptic curve types not supported
+  throw new Error("unsupported modulus value");
+};
+
+/**
+ * Compute the modular exponentiation.
+ */
+const modPow = (b: bigint, exp: bigint, p: bigint): bigint => {
+  if (exp === BigInt(0)) {
+    return BigInt(1);
+  }
+  let result = b;
+  const exponentBitString = exp.toString(2);
+  for (let i = 1; i < exponentBitString.length; ++i) {
+    result = (result * result) % p;
+    if (exponentBitString[i] === "1") {
+      result = (result * b) % p;
+    }
+  }
+  return result;
+};
+
+/**
+ * Test if a specific bit is set.
+ */
+const testBit = (n: bigint, i: number): boolean => {
+  const m = BigInt(1) << BigInt(i);
+  return (n & m) !== BigInt(0);
+};
+
+/**
+ * Generate a random Uint8Array of a specific length.
+ */
+const randomBytes = (length: number): Uint8Array => {
+  const array = new Uint8Array(length);
+  for (let i = 0; i < length; i++) {
+    array[i] = Math.floor(Math.random() * 256);
+  }
+  return array;
+};
+
+/**
+ * Create additional associated data (AAD) for AES-GCM decryption.
+ */
+const additionalAssociatedData = (
+  senderPubBuf: ArrayBuffer,
+  receiverPubBuf: ArrayBuffer
+): Uint8Array => {
+  const s = Array.from(new Uint8Array(senderPubBuf));
+  const r = Array.from(new Uint8Array(receiverPubBuf));
+  return new Uint8Array([...s, ...r]);
+};
+
+/**
+ * Perform HKDF extract and expand operations.
+ */
+const extractAndExpand = async (
+  sharedSecret: Uint8Array,
+  ikm: Uint8Array,
+  info: Uint8Array,
+  len: number
+): Promise<Uint8Array> => {
+  const prk = hkdf.extract(sha256, ikm, sharedSecret);
+  const resp = hkdf.expand(sha256, prk, info, len);
+  return new Uint8Array(resp);
+};
+
+/**
+ * Derive the Diffie-Hellman (DH) shared secret using ECDH.
+ */
+const deriveDh = async (
+  encappedKeyBuf: Uint8Array,
+  receiverPriv: string
+): Promise<Uint8Array> => {
+  const dh = p256.getSharedSecret(
+    base64urlDecode(receiverPriv),
+    encappedKeyBuf
+  );
+  return dh.slice(1);
+};
+
+/**
+ * Decrypt data using AES-GCM.
+ */
+const aesGcmDecrypt = (
+  encryptedData: Uint8Array,
+  key: Uint8Array,
+  iv: Uint8Array,
+  aad?: Uint8Array
+): Uint8Array => {
+  const aes = gcm(key, iv, aad);
+  const data_ = aes.decrypt(encryptedData);
+  return data_;
+};
+
+/**
+ * Generate a Key Encapsulation Mechanism (KEM) context.
+ */
+const getKemContext = (
+  encappedKeyBuf: Uint8Array,
+  publicKey: string
+): Uint8Array => {
+  const encappedKeyArray = new Uint8Array(encappedKeyBuf);
+  const publicKeyArray = uint8arrayFromHexString(publicKey);
+
+  const kemContext = new Uint8Array(
+    encappedKeyArray.length + publicKeyArray.length
+  );
+  kemContext.set(encappedKeyArray);
+  kemContext.set(publicKeyArray, encappedKeyArray.length);
+
+  return kemContext;
+};
```

Comment:
At the end of the day, this function is doing a simple concatenation:
```js
new Uint8Array([...encappedKeyBuf, ...uint8arrayFromHexString(publicKey)])
```

Let's inline it given it's only used once?